src/Components/animation/Animation.js

```diff
@@ -134,7 +137,26 @@ class Animation extends React.Component {
   render() {
     return (
       <View>
-        <View style={{ height: this.animationHeight, width: this.animationWidth }}>
+        <View
+          ref={ref => {
+            this.marker = ref;
+          }}
+          onLayout={({ nativeEvent }) => {
+            if (this.marker) {
+              this.marker.measure((x, y, width, height, pageX, pageY) => {
+                // On iOS, when the left margin is = 0, pageX can be equal to the whole width instead of 0
+                if (pageX > 0.99 * width) pageX = 0;
+
+                this.dTop = pageY;
+                this.dLeft = pageX;
+
+                if (this.props.onTopMarginSet !== undefined && this.props.onTopMarginSet !== null)
+                  this.props.onTopMarginSet(this.dTop);
```

Comment:
It seems a bit unclear to me who is responsible for the layout size, I'll do some tests to understand what's happening both in the editor and in a drill

Reply:
- If your question is why I am doing this, here is some context:
In this code, the animation is sending its position to the editor.

This is used by the editor, when an element is dropped on the screen. Thanks to this information, the editor can compute the position of the element in the animation.

- If your question is who decides how the layout is done:
I don't exactly know how a layout is set in react-native but I think that it is done as usual here (I do not use any trick to change the layout). The only difference is that once the Animation is set, it sends its position to its parent (but this position is not used to influence the layout, just the position of the elements that will be added to the animation in the future).

Reply:
The thing is we're using all this code that is part of Animation to fix something in the Editor, which is a bit weird because it is going to be called also on the DrillPage because this is used everywhere an Animation is displayed.

Reply:
I don't see how to make the editor precisely know where an element is dropped otherwise.